app/models/docker_builder_service.rb

```diff
@@ -8,6 +8,28 @@ class DockerBuilderService
 
   attr_reader :build, :execution
 
+  def self.build_docker_image(dir, docker_options, output)
+    output.puts("### Running Docker build")
+
+    docker_image =
+      Docker::Image.build_from_dir(dir, docker_options, Docker.connection, registry_credentials) do |chunk|
+        output.write_docker_chunk(chunk)
+      end
+    output.puts('### Docker build complete')
+
+    docker_image
+  end
+
+  def self.registry_credentials
+    return nil unless ENV['DOCKER_REGISTRY'].present?
```

Comment:
could do `return unless registry = ENV['DOCKER_REGISTRY'].presence` and then reuse registry below